ci/jenkins/utils.sh

```diff
@@ -0,0 +1,62 @@
+#!/usr/bin/env bash
+
+# Copyright 2023 Antrea Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+function check_and_upgrade_golang() {
+    export GOROOT=/usr/local/go
+    export PATH=${GOROOT}/bin:$PATH
+    echo "====== Checking installed Golang version ======"
+    # Since golang release 1.21, the version format in go.mod is changed from "go 1.21" to "go1.21.0".
+    # But if the golang version on testbed is older than 1.21, command "go mod edit -json | jq -r '.Go'"
+    # will return an error. So we using the following command to get the golang version.
+    antrea_golang_version="go$(grep -E "^go\ " go.mod | awk '{print $2}')"
+    golang_minor_version=$(echo ${antrea_golang_version} | awk -F '.' '{print $2}')
+    # If Antrea Golang version is higher than 1.21, and go.mod does not contains the patch number
+    # (e.g. go 1.21), we need to add the patch number to the golang version (e.g. go1.21.0).
+    if [[ "${golang_minor_version}" -ge 21 ]] && [[ ! "${antrea_golang_version}" =~ ^go[0-9]+\.[0-9]+\.[0-9]+$ ]]; then
+        antrea_golang_version="${antrea_golang_version}.0"
+    fi
```

Comment:
I don't understand why the code got so complicated. All we need is to find out the go minor version from the go.mod (knowing that the go directive in the go.mod can be of either form: `X.Y.Z` or `X.Y`). Then we need to download and install the most recent patch release for that minor version.

In a way, this is basically what you had before, but we need to accept both forms of go directives in the go.mod.

Reply:
In the previous implementation, both `go 1.21` and `go 1.21.1` is acceptable,  if the golang version is in format `go X.Y`, this script will use the first version of the major release, for example `go 1.21 ==> go 1.21.0`.  But if the golang version is older than `go 1.21`, which means the first minor version is in format `go X.Y`, such `go 1.20`, then this script will use this version directly. That's why I checked the minor version.

But I think use the most recent version is more reasonable, I removed this check now. In current implementation, if patch number is specified, this script will use the accurate version in `go.mod`, otherwise it will use the most recent patch release. Thanks for reminder.